examples/fp8_attention.py

```diff
@@ -0,0 +1,212 @@
+
+            # Load query tile - keep in FP8
+            q_tile = q[bh, tile_m, :]  # [tile_m, dim]
+
+            # Compute attention scores for all keys
+            for tile_n in hl.tile(seq_len):
+                # Load key tile and transpose for Q @ K^T
+                k_tile = k[bh, tile_n, :]  # [tile_n, dim] - keep in FP8
+                k_tile_t = k_tile.transpose(0, 1)  # [dim, tile_n]
+
+                # Compute Q @ K^T with FP8 inputs, result in FP32
+                qk = torch.matmul(q_tile, k_tile_t).to(
+                    torch.float32
+                )  # [tile_m, tile_n]
+
+                # Scale QK scores first
+                qk_scaled = qk * sm_scale  # [tile_m, tile_n]
+
+                # Compute max of scaled scores
+                qk_max = torch.amax(qk_scaled, dim=-1)  # [tile_m]
+
+                # Update global max
+                m_new = torch.maximum(m_i, qk_max)
+
+                # Shift by max for numerical stability
+                qk_shifted = qk_scaled - m_new[:, None]
+
+                # Use exp2 to match Triton kernel's implementation
+                # Note: Triton kernel already multiplies sm_scale by 1.44269504
+                p = torch.exp2(qk_shifted)  # [tile_m, tile_n]
+
+                # Sum of exponentials for this block
+                l_ij = torch.sum(p, dim=-1)  # [tile_m]
+
+                # Update accumulators with correction factor
+                # Correction factor for previous blocks
+                alpha = torch.exp2(m_i - m_new)
+                l_i = l_i * alpha + l_ij
+                acc = acc * alpha[:, None]
+
+                # Load values - V is [dim, seq]
+                v_tile = v[bh, :, tile_n]  # [dim, tile_n] - keep in FP8
+
+                # Convert p to FP8 for FP8 GEMM
+                p_fp8 = p.to(v.dtype)  # Convert to same FP8 type as V
+
+                # Accumulate attention @ V with FP8 GEMM
+                v_t = v_tile.transpose(0, 1)  # [tile_n, dim]
+                pv = torch.matmul(p_fp8, v_t).to(torch.float32)  # [tile_m, dim]
+                acc = acc + pv
+
+                # Update max tracker
+                m_i = m_new
+
+            # Final normalization
+            acc = acc / l_i[:, None]
+            out[bh, tile_m, :] = acc
+
+    return out
+
+
+def preprocess_fp8_attention_inputs(
+    q: torch.Tensor, k: torch.Tensor, v: torch.Tensor
+) -> tuple[torch.Tensor, torch.Tensor, torch.Tensor]:
+    q_fp8 = q.to(torch.float8_e5m2)
+    k_fp8 = k.to(torch.float8_e5m2)
+    v = v.permute(0, 1, 3, 2)
+    v_fp8 = v.to(torch.float8_e5m2)
+    batch, heads, seq_len, head_dim = q.shape
+    q_fp8_reshaped = q_fp8.reshape(batch * heads, seq_len, head_dim)
+    k_fp8_reshaped = k_fp8.reshape(batch * heads, seq_len, head_dim)
+    v_fp8_reshaped = v_fp8.reshape(batch * heads, head_dim, seq_len)
+    return q_fp8_reshaped, k_fp8_reshaped, v_fp8_reshaped
+
+
+def fp8_attention_tritonbench(
```

Comment:
Remove this?  I don't understand why this is needed.

Reply:
It matches the structure of other tritonbench kernels (running preprocessing to convert input from fp16 to fp8, but exclude preprocessing from kernel timing measurement):

https://github.com/pytorch-labs/tritonbench/blob/91cc742c54fc25f1dc4a5bb1950b350e6f641ce9/tritonbench/operators/fp8_attention/operator.py#L122-L133

```python
    @register_benchmark(baseline=True)
    def triton_flash_v2(
        self,
        q: torch.Tensor,
        k: torch.Tensor,
        v: torch.Tensor,
    ) -> Callable:
        triton_q, triton_k, triton_v = self.triton_preprocess(q, k, v)
        # full fp8 will be enabled if type of q,k,v is fp8
        return lambda: triton_attention(
            triton_q, triton_k, triton_v, self.causal, self.sm_scale, "base_opt"
        )
```

Reply:
Also updated the pytorch reference impl to use the same setup